From 63a1933ed8eb04a1b7924238710b8740916509bd Mon Sep 17 00:00:00 2001
From: Sebastian Bazley <sebb@apache.org>
Date: Sun, 4 Jan 2015 21:03:42 +0000
Subject: [PATCH] VALIDATOR-350 - Should "x.root" validate as a domain name?
 Removed "root" from TLD list. Also "um" and "yu" as they are currently "Not
 assigned"

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/validator/trunk@1649419 13f79535-47bb-0310-9956-ffa450edef68
---
 RELEASE-NOTES.txt                                           | 4 +++-
 src/changes/changes.xml                                     | 5 +++++
 .../apache/commons/validator/routines/DomainValidator.java  | 6 ------
 3 files changed, 8 insertions(+), 7 deletions(-)

diff --git a/RELEASE-NOTES.txt b/RELEASE-NOTES.txt
index 40ab1e833..01f96be4e 100644
--- a/RELEASE-NOTES.txt
+++ b/RELEASE-NOTES.txt
@@ -74,6 +74,9 @@ BUGS FROM PREVIOUS RELEASE
  * [VALIDATOR-308] - Logical errors in util.Flags affecting check of multiple
                      flags as well as flag 64
                      Thanks to Til Boerner
+ * [VALIDATOR-350] - Should "x.root" validate as a domain name?
+                     Removed "root" from TLD list. 
+                     Also "um" and "yu" as they are currently "Not assigned" 
 
 IMPROVEMENTS OVER PREVIOUS RELEASE
 ===================================
@@ -88,7 +91,6 @@ IMPROVEMENTS OVER PREVIOUS RELEASE
                      Fix up the testCalculateInvalid() invalid method to allow for 
                      either invalid checksum or syntax (CheckDigitException) error
                      when testing the entries in the invalid array.
-                     
 
 DEPRECATIONS
 ============
diff --git a/src/changes/changes.xml b/src/changes/changes.xml
index d2984bd63..f69ab466f 100644
--- a/src/changes/changes.xml
+++ b/src/changes/changes.xml
@@ -43,6 +43,11 @@ The <action> type attribute can be add,update,fix,remove.
   <body>
 
   <release version="1.4.1" date="tba" description="1.4 Maintenance release">
+    <action issue="VALIDATOR-350" dev="sebb" type="fix">
+     Should "x.root" validate as a domain name?
+     Removed "root" from TLD list. 
+     Also "um" and "yu" as they are currently "Not assigned" 
+    </action>
     <action issue="VALIDATOR-308" dev="sebb" type="fix" due-to="Til Boerner">
       Logical errors in util.Flags affecting check of multiple flags as well as flag 64
     </action>
diff --git a/src/main/java/org/apache/commons/validator/routines/DomainValidator.java b/src/main/java/org/apache/commons/validator/routines/DomainValidator.java
index 2f0a9dfd2..be79f6a65 100644
--- a/src/main/java/org/apache/commons/validator/routines/DomainValidator.java
+++ b/src/main/java/org/apache/commons/validator/routines/DomainValidator.java
@@ -232,7 +232,6 @@ private String chompLeadingDot(String str) {
 
     private static final String[] INFRASTRUCTURE_TLDS = new String[] {
         "arpa",               // internet infrastructure
-        "root"                // diagnostic marker for non-truncated root zone
     };
 
     private static final String[] GENERIC_TLDS = new String[] {
@@ -1013,8 +1012,6 @@ private String chompLeadingDot(String str) {
         "ua",                 // Ukraine
         "ug",                 // Uganda
         "uk",                 // United Kingdom
-        "um",                 // United States Minor Outlying Islands
-                              // TODO um is not in the IANA text file, it is in the HTML file
         "us",                 // United States of America
         "uy",                 // Uruguay
         "uz",                 // Uzbekistan
@@ -1067,9 +1064,6 @@ private String chompLeadingDot(String str) {
         "xn--ygbi2ammx", // فلسطين Ministry of Telecom &amp; Information Technology (MTIT)
         "ye",                 // Yemen
         "yt",                 // Mayotte
-        "yu",                 // Serbia and Montenegro (originally Yugoslavia)
-                              // TODO yu is not in the IANA text file, nor it is in the HTML file
-                              // It looks like it has been removed from use
         "za",                 // South Africa
         "zm",                 // Zambia
         "zw",                 // Zimbabwe
